packages/legend-query-builder/src/components/result/tds/QueryBuilderTDSGridResult.tsx

```diff
@@ -169,19 +207,28 @@ export const QueryBuilderTDSGridResult = observer(
             />
           ) : (
             <DataGrid
-              rowData={rowData}
+              rowData={getRowDataFromExecutionResult(executionResult)}
               gridOptions={{
                 suppressScrollOnNewData: true,
                 getRowId: (data) => data.data.rowNumber,
                 rowSelection: 'multiple',
+                enableRangeSelection: true,
               }}
               // NOTE: when column definition changed, we need to force refresh the cell to make sure the cell renderer is updated
               // See https://stackoverflow.com/questions/56341073/how-to-refresh-an-ag-grid-when-a-change-occurs-inside-a-custom-cell-renderer-com
               onRowDataUpdated={(params) => {
                 params.api.refreshCells({ force: true });
               }}
+              onRangeSelectionChanged={(event) => {
+                const selectedCells = getSelectedCells(event.api);
+                resultState.setSelectedCells([]);
+                selectedCells.forEach((cell) =>
+                  resultState.addSelectedCell(cell),
+                );
+              }}
               suppressFieldDotNotation={true}
-              suppressContextMenu={true}
+              suppressClipboardPaste={false}
+              suppressContextMenu={!isAdvancedModeEnabled}
```

Comment:
why are we suppressing the context menu that ag grid provides. cant we leverage what they give us and just configure it to our use case ? 
https://codesandbox.io/s/ycphfx?file=/main.js
https://www.ag-grid.com/javascript-data-grid/context-menu/
Seems like we just add the filter by/ filter out handlers and use their menu